backend/review/views.py

```diff
@@ -125,39 +124,95 @@ def extra_metrics_section_filters_pcr(current_semester=None):
                 }
             },
         },
+        custom_parameters={
+            "course-reviews": {
+                "GET": [
+                    {
+                        "name": "semester",
+                        "in": "query",
+                        "description": "Optionally specify the semester of the desired course (defaults to most recent course with the specified course code).",  # noqa E501
+                        "schema": {"type": "string"},
+                        "required": False,
+                    },
+                ]
+            },
+        },
         override_response_schema=course_reviews_response_schema,
     )
 )
 @permission_classes([IsAuthenticated])
-def course_reviews(request, course_code):
+def course_reviews(request, course_code, semester=None):
     """
     Get all reviews for the topic of a given course and other relevant information.
     Different aggregation views are provided, such as reviews spanning all semesters,
     only the most recent semester, and instructor-specific views.
     """
     try:
+        semester = request.GET.get("semester")
         course = (
-            Course.objects.filter(course_filters_pcr_allow_xlist, full_code=course_code)
+            Course.objects.filter(
+                course_filters_pcr,
+                **(
+                    {"topic__courses__full_code": course_code, "topic__courses__semester": semester}
+                    if semester
+                    else {"full_code": course_code}
+                ),
```

Comment:
I'm not understanding why we don't use the topic when the semester is not specified. Would you mind explaining?

Reply:
If the semester isn't specified, we just take the most recent course matching the full code. If the semester is specified, we find the most recent course in the same topic as a course matching the given full code in the given semester.